### Análise Exploratória dos Dados de Missões Espaciais realizadas a partir de 1957

Alunas: Fernanda da Silva Leite e 

O aplicativo deste trabalho está disponível no [Streamlit](https://trabalho-intcd.streamlit.app/).

In [28]:
# Importação das Bibliotecas
import pandas as pd
import plotly.express as px
import matplotlib as mp
import numpy as np
import seaborn as sns

In [29]:
# Importação dos Dados - Criação do Dataframe e leitura dos dados do arquivo CSV
df = pd.read_csv("data/Space_Corrected.csv")

In [30]:
# Visualização dos Dados
df.head()

Unnamed: 0.1  Unnamed: 0 Company Name  \
0             0           0       SpaceX   
1             1           1         CASC   
2             2           2       SpaceX   
3             3           3    Roscosmos   
4             4           4          ULA   

                                            Location  \
0         LC-39A, Kennedy Space Center, Florida, USA   
1  Site 9401 (SLS-2), Jiuquan Satellite Launch Ce...   
2                      Pad A, Boca Chica, Texas, USA   
3       Site 200/39, Baikonur Cosmodrome, Kazakhstan   
4           SLC-41, Cape Canaveral AFS, Florida, USA   

                        Datum                                        Detail  \
0  Fri Aug 07, 2020 05:12 UTC  Falcon 9 Block 5 | Starlink V1 L9 & BlackSky   
1  Thu Aug 06, 2020 04:01 UTC           Long March 2D | Gaofen-9 04 & Q-SAT   
2  Tue Aug 04, 2020 23:57 UTC            Starship Prototype | 150 Meter Hop   
3  Thu Jul 30, 2020 21:25 UTC  Proton-M/Briz-M | Ekspress-80 & Ekspress-103   
4  Thu Jul 30, 2020 11:50 UTC                    Atlas V 541 | Perseverance   

  Status Rocket  Rocket Status Mission  
0  StatusActive   50.0         Success  
1  StatusActive  29.75         Success  
2  StatusActive     NaN        Success  
3  StatusActive   65.0         Success  
4  StatusActive  145.0         Success

In [31]:
# Informações do Dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4324 entries, 0 to 4323
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Unnamed: 0.1    4324 non-null   int64 
 1   Unnamed: 0      4324 non-null   int64 
 2   Company Name    4324 non-null   object
 3   Location        4324 non-null   object
 4   Datum           4324 non-null   object
 5   Detail          4324 non-null   object
 6   Status Rocket   4324 non-null   object
 7    Rocket         964 non-null    object
 8   Status Mission  4324 non-null   object
dtypes: int64(2), object(7)
memory usage: 304.2+ KB


In [32]:
# Visualizando se há valores ausentes
df.isnull().sum()

Unnamed: 0.1         0
Unnamed: 0           0
Company Name         0
Location             0
Datum                0
Detail               0
Status Rocket        0
 Rocket           3360
Status Mission       0
dtype: int64

In [33]:
# Remove colunas desnecessárias
df.drop(columns=['Unnamed: 0', 'Unnamed: 0.1'], inplace=True)

# Estatísticas Descritivas das variáveis numéricas
df.describe()

Company Name                                    Location  \
count          4324                                        4324   
unique           56                                         137   
top       RVSN USSR  Site 31/6, Baikonur Cosmodrome, Kazakhstan   
freq           1777                                         235   

                             Datum                               Detail  \
count                         4324                                 4324   
unique                        4319                                 4278   
top     Wed Nov 05, 2008 00:15 UTC  Cosmos-3MRB (65MRB) | BOR-5 Shuttle   
freq                             2                                    6   

        Status Rocket  Rocket Status Mission  
count            4324     964           4324  
unique              2      56              4  
top     StatusRetired  450.0         Success  
freq             3534     136           3879

---

#### Informações sobre os Dados

O arquivo é composto por 4.324 linhas e 9 colunas, sendo elas:
- `Unnamed: 0.1` - Coluna com index, que será removida posteriormente. Possui 4.324 registros não-nulos;
- `Unnamed: 0` - Coluna com index, que será removida posteriormente. Possui 4.324 registros não-nulos; 
- `Company Name` (Nomes das Empresas) - Coluna que apresenta os nomes das empresas que realizaram missões espaciais. Possui 4.324 registros não-nulos;
- `Location` (Localização) - Coluna que apresenta o local em que foram realizadas missões espaciais, incluindo país, cidade e base de lançamento. Possui 4.324 registros não-nulos; 
- `Datum` (Data) - Coluna que apresenta a data e o horário de lançamento do foguete. Possui 4.324 registros não-nulos; 
- `Detail` (Detalhe) - Coluna que apresenta o nome do foguete. Possui 4.234 registros não-nulos;
- `Status Rocket` (Estatus do Foguete) - Coluna que apresenta a situação do foguete como Ativo ou Aposentado. Possui 4.324 registros não-nulos; 
- `Rocket` (Foguete) - Coluna que apresenta o custo da missão em milhões de dólares. Possui 964 registros não-nulos; 
- `Status` Mission (Status da Missão) - Coluna que apresenta a situação da missão como Sucedida ou Fracassada. Possui 4.324 registros não-nulos.

---

In [34]:
# Company Name - Visualiza quais empresas únicas estão no DataFrame
empresas = df['Company Name'].unique()
empresas

array(['SpaceX', 'CASC', 'Roscosmos', 'ULA', 'JAXA', 'Northrop', 'ExPace',
       'IAI', 'Rocket Lab', 'Virgin Orbit', 'VKS RF', 'MHI', 'IRGC',
       'Arianespace', 'ISA', 'Blue Origin', 'ISRO', 'Exos', 'ILS',
       'i-Space', 'OneSpace', 'Landspace', 'Eurockot', 'Land Launch',
       'CASIC', 'KCST', 'Sandia', 'Kosmotras', 'Khrunichev', 'Sea Launch',
       'KARI', 'ESA', 'NASA', 'Boeing', 'ISAS', 'SRC', 'MITT', 'Lockheed',
       'AEB', 'Starsem', 'RVSN USSR', 'EER', 'General Dynamics',
       'Martin Marietta', 'Yuzhmash', 'Douglas', 'ASI', 'US Air Force',
       'CNES', 'CECLES', 'RAE', 'UT', 'OKB-586', 'AMBA',
       "Arm??e de l'Air", 'US Navy'], dtype=object)

In [35]:
# Location - Visualiza quantos locais de lançamento únicos existem
locais = df['Location'].unique()
print(f"Total de locais únicos: {len(locais)}")

Total de locais únicos: 137


In [36]:
# Cria uma nova coluna chamada Country com apenas o nome do país onde o lançamento foi realizado.
# str.split(',') → divide a string em partes onde tem vírgulas
# str[-1] → pega a última parte (que é o país)
# str.strip() → remove espaços em branco extras
df['Country'] = df['Location'].str.split(',').str[-1].str.strip()

df.head(3)

Company Name                                           Location  \
0       SpaceX         LC-39A, Kennedy Space Center, Florida, USA   
1         CASC  Site 9401 (SLS-2), Jiuquan Satellite Launch Ce...   
2       SpaceX                      Pad A, Boca Chica, Texas, USA   

                        Datum                                        Detail  \
0  Fri Aug 07, 2020 05:12 UTC  Falcon 9 Block 5 | Starlink V1 L9 & BlackSky   
1  Thu Aug 06, 2020 04:01 UTC           Long March 2D | Gaofen-9 04 & Q-SAT   
2  Tue Aug 04, 2020 23:57 UTC            Starship Prototype | 150 Meter Hop   

  Status Rocket  Rocket Status Mission Country  
0  StatusActive   50.0         Success     USA  
1  StatusActive  29.75         Success   China  
2  StatusActive     NaN        Success     USA

In [37]:
# Na coluna 'Datum', remove o horário e deixa apenas a data. Ex: 'Fri Aug 07, 2020 05:00 UTC' se torna 'Fri Aug 07, 2020'
df['Datum'] = df['Datum'].str.split(' ').str[:4].str.join(' ')
df

Company Name                                           Location  \
0          SpaceX         LC-39A, Kennedy Space Center, Florida, USA   
1            CASC  Site 9401 (SLS-2), Jiuquan Satellite Launch Ce...   
2          SpaceX                      Pad A, Boca Chica, Texas, USA   
3       Roscosmos       Site 200/39, Baikonur Cosmodrome, Kazakhstan   
4             ULA           SLC-41, Cape Canaveral AFS, Florida, USA   
...           ...                                                ...   
4319      US Navy           LC-18A, Cape Canaveral AFS, Florida, USA   
4320         AMBA           LC-26A, Cape Canaveral AFS, Florida, USA   
4321      US Navy           LC-18A, Cape Canaveral AFS, Florida, USA   
4322    RVSN USSR          Site 1/5, Baikonur Cosmodrome, Kazakhstan   
4323    RVSN USSR          Site 1/5, Baikonur Cosmodrome, Kazakhstan   

                 Datum                                        Detail  \
0     Fri Aug 07, 2020  Falcon 9 Block 5 | Starlink V1 L9 & BlackSky   
1     Thu Aug 06, 2020           Long March 2D | Gaofen-9 04 & Q-SAT   
2     Tue Aug 04, 2020            Starship Prototype | 150 Meter Hop   
3     Thu Jul 30, 2020  Proton-M/Briz-M | Ekspress-80 & Ekspress-103   
4     Thu Jul 30, 2020                    Atlas V 541 | Perseverance   
...                ...                                           ...   
4319  Wed Feb 05, 1958                     Vanguard | Vanguard TV3BU   
4320  Sat Feb 01, 1958                           Juno I | Explorer 1   
4321  Fri Dec 06, 1957                       Vanguard | Vanguard TV3   
4322  Sun Nov 03, 1957                    Sputnik 8K71PS | Sputnik-2   
4323  Fri Oct 04, 1957                    Sputnik 8K71PS | Sputnik-1   

      Status Rocket  Rocket Status Mission     Country  
0      StatusActive   50.0         Success         USA  
1      StatusActive  29.75         Success       China  
2      StatusActive     NaN        Success         USA  
3      StatusActive   65.0         Success  Kazakhstan  
4      StatusActive  145.0         Success         USA  
...             ...     ...            ...         ...  
4319  StatusRetired     NaN        Failure         USA  
4320  StatusRetired     NaN        Success         USA  
4321  StatusRetired     NaN        Failure         USA  
4322  StatusRetired     NaN        Success  Kazakhstan  
4323  StatusRetired     NaN        Success  Kazakhstan  

[4324 rows x 8 columns]

In [38]:
# Separa a coluna 'Datum' em duas colunas: 'Weekday' e 'Date
df[['Weekday', 'Date']] = df['Datum'].str.split(' ', n=1, expand=True)

# Remove a coluna 'Datum' do DataFrame, pois agora temos as informações necessárias em 'Weekday' e 'Date'
df.drop(columns=['Datum'], inplace=True)

In [39]:
# Remove espaços em branco extras nas colunas 'Weekday' e 'Date'
df['Weekday'] = df['Weekday'].str.strip()
df['Date'] = df['Date'].str.strip()

# Converte a coluna 'Date' para o tipo datetime
df['Date'] = pd.to_datetime(df['Date'])

C:\Users\ferna\AppData\Local\Temp\ipykernel_18600\3451947417.py:6: UserWarning:

Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.



---

## Análise Exploratória

#### Quantidade de Lançamentos por País

In [40]:
# Quantidade de lançamentos em cada país
lancamentos_pais = df['Country'].value_counts()

# Transforma as localidades em Países, unificando alguns locais que pertencem ao mesmo país.
replace_dict = {
    'Barents Sea': 'Russia',
    'Gran Canaria': 'Spain',
    'Shahrud Missile Test Site': 'Iran',
    'Yellow Sea': 'Russia',
    'Pacific Missile Range Facility': 'USA',
    'Pacific Ocean': 'USA'
}

# Substitui os nomes na Series original
lancamentos_pais_corrigido = lancamentos_pais.rename(index=replace_dict)

# Agrupa novamente para somar os valores dos países que foram unificados
lancamentos_pais = lancamentos_pais_corrigido.groupby(lancamentos_pais_corrigido.index).sum()

# Ordena os valores de lançamentos por país em ordem decrescente
lancamentos_pais = lancamentos_pais.sort_values(ascending=False)

lancamentos_pais

Country
Russia         1399
USA            1381
Kazakhstan      701
France          303
China           268
Japan           126
India            76
Iran             14
New Zealand      13
Israel           11
Kenya             9
Australia         6
North Korea       5
New Mexico        4
Brazil            3
South Korea       3
Spain             2
Name: count, dtype: int64

In [41]:
# Criação de um gráfico de barras com a quantidade de lançamentos por país, 10 países com mais lançamentos
fig = px.bar(lancamentos_pais.head(10), 
             x=lancamentos_pais.head(10).index, 
             y=lancamentos_pais.head(10).values, 
             title='Top 10 Países com mais lançamentos espaciais',
             labels={'x': 'País', 'y': 'Quantidade de Lançamentos'})
fig.update_layout(xaxis_title='País', yaxis_title='Quantidade de Lançamentos')
# Mostra os valores no topo de cada barra
fig.update_traces(texttemplate='%{y}', textposition='outside')
fig.show()

Este gráfico resume a contribuição histórica acumulada de cada nação para a atividade de lançamento. A Rússia (ex-URSS) e os EUA dominam a lista com 1399 e 1381 lançamentos, respectivamente, reafirmando sua primazia histórica e o impacto da Guerra Fria na aceleração das missões espaciais. O Cazaquistão (701) aparece em terceiro devido ao uso do Cosmódromo de Baikonur, a primeira e maior base de lançamentos de foguetes do mundo. A França e a China seguem com números significativos (303 e 268), destacando suas capacidades espaciais bem estabelecidas, enquanto Japão e Índia também mostram um número considerável. A presença de Irã, Nova Zelândia e Israel com volumes menores indica a expansão do "clube espacial" para incluir países com programas mais recentes ou de nicho. Este gráfico é uma representação concisa da concentração de capacidade de lançamento espacial nas mãos de poucas potências ao longo da história

#### Em relação a Data

In [42]:
ano_mais_lancamentos = df['Date'].dt.year.value_counts().idxmax()
print(f"O ano com mais lançamentos espaciais foi {ano_mais_lancamentos}.")

O ano com mais lançamentos espaciais foi 1971.


In [43]:
# Gráfico com o número de lançamentos por ano, individualizando os anos
fig_ano = px.bar(df['Date'].dt.year.value_counts().sort_index(), 
                 x=df['Date'].dt.year.value_counts().sort_index().index, 
                 y=df['Date'].dt.year.value_counts().sort_index().values, 
                 title='Número de Lançamentos Espaciais por Ano',
                 labels={'x': 'Ano', 'y': 'Número de Lançamentos'})
fig_ano.update_layout(xaxis_title='Ano', yaxis_title='Número de Lançamentos')
# Separa as barras
fig_ano.update_traces(marker=dict(line=dict(width=1, color='DarkSlateGrey')))
fig_ano.update_xaxes(dtick="M12", tickformat="%Y")  # Formata o eixo x para mostrar apenas o ano
fig_ano.show()


O gráfico oferece uma perspectiva histórica sobre a frequência dos lançamentos. Observa-se um rápido crescimento nos anos 1960 e 1970, impulsionado pela corrida espacial entre EUA e URSS, com picos significativos por volta de 1967-1968 e novamente em meados dos anos 1970 (1975-1976), quando ambos os países realizavam dezenas de lançamentos anuais. Após esses picos, houve uma ligeira diminuição na frequência nos anos 1980 e 1990, e uma estabilização nos anos 2000. No entanto, a década de 2010 mostra um ressurgimento notável no número de lançamentos, com um novo pico em 2018 (quase 120 lançamentos), impulsionado pela ascensão de novos players como a China, o crescimento de empresas privadas e o aumento da demanda por serviços de lançamento de satélites, indicando uma nova era de expansão espacial.

In [44]:
# Em relação aos meses, o mês com mais lançamentos espaciais foi:
mes_mais_lancamentos = df['Date'].dt.month.value_counts().idxmax()
print(f"O mês com mais lançamentos espaciais foi {mes_mais_lancamentos}.")

# Gráfico com o número de lançamentos por mês, individualizando os meses
fig_mes = px.bar(df['Date'].dt.month.value_counts().sort_index(), 
                 x=df['Date'].dt.month.value_counts().sort_index().index, 
                 y=df['Date'].dt.month.value_counts().sort_index().values, 
                 title='Número de Lançamentos Espaciais por Mês',
                 labels={'x': 'Mês', 'y': 'Número de Lançamentos'})
fig_mes.update_layout(xaxis_title='Mês', yaxis_title='Número de Lançamentos')
# Separa as barras
fig_mes.update_traces(marker=dict(line=dict(width=1, color='DarkSlateGrey')))
fig_mes.update_xaxes(tickvals=np.arange(1, 13), 
                     ticktext=['Jan', 'Fev', 'Mar', 'Abr', 'Mai', 'Jun', 
                               'Jul', 'Ago', 'Set', 'Out', 'Nov', 'Dez'])  # Formata o eixo x para mostrar os meses
fig_mes.show()

O mês com mais lançamentos espaciais foi 12.


O gráfico revela uma distribuição relativamente uniforme de lançamentos ao longo do ano, com uma leve tendência de aumento no final do ano. Dezembro se destaca como o mês com o maior número de lançamentos (450), seguido de perto por abril, junho e outubro. Essa distribuição sugere que não há sazonalidade climática ou operacional rígida que concentre lançamentos em apenas alguns meses, indicando que as janelas de lançamento são aproveitadas de forma consistente, possivelmente otimizando o uso das infraestruturas e atendendo a demandas de órbita específicas ou a prazos de projetos que se estendem ao longo de todo o ano.

In [45]:
# Dia da Semana - O dia da semana com mais lançamentos espaciais foi:
dia_mais_lancamentos = df['Weekday'].value_counts().idxmax()
print(f"O dia da semana com mais lançamentos espaciais foi {dia_mais_lancamentos}.")
# Gráfico com o número de lançamentos por dia da semana, individualizando os dias
fig_dia = px.bar(df['Weekday'].value_counts().sort_index(), 
                 x=df['Weekday'].value_counts().sort_index().index, 
                 y=df['Weekday'].value_counts().sort_index().values, 
                 title='Número de Lançamentos Espaciais por Dia da Semana',
                 labels={'x': 'Dia da Semana', 'y': 'Número de Lançamentos'})
fig_dia.update_layout(xaxis_title='Dia da Semana', yaxis_title='Número de Lançamentos')
# Separa as barras
fig_dia.update_traces(marker=dict(line=dict(width=1, color='DarkSlateGrey')))
fig_dia.update_xaxes(tickvals=['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'], 
                     ticktext=['Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sáb', 'Dom'])  # Formata o eixo x para mostrar os dias da semana
# Ordena os dias da semana na ordem correta
fig_dia.update_xaxes(categoryorder='array', categoryarray=['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'])

fig_dia.show()


O dia da semana com mais lançamentos espaciais foi Wed.


Este gráfico demonstra a frequência de lançamentos espaciais distribuída pelos dias da semana. Notavelmente, a maioria dos lançamentos ocorre nos dias úteis, com quartas, quintas e sextas-feiras apresentando o maior número de atividades. Há uma queda acentuada nos lançamentos durante os fins de semana, especialmente aos domingos, que mostram a menor quantidade. Essa distribuição reflete as operações típicas de uma indústria que segue um cronograma de trabalho padronizado, onde a disponibilidade de equipes, infraestrutura e apoio técnico é maior durante a semana comercial, priorizando a eficiência e coordenação das operações.

#### Sucesso das Missões

In [46]:
missoes_sucesso = df[df['Status Mission'] == 'Success'].shape[0]

# Contagem de missões com status de Failure, Partial Failure e Prelaunch Failure
missoes_falha = df[df['Status Mission'].isin(['Failure', 'Partial Failure', 'Prelaunch Failure'])].shape[0]

# Cria uma nova coluna 'Mission Status' para categorizar as missões
df['Mission Status'] = df['Status Mission'].apply(lambda x: 'Success' if x == 'Success' else 'Failure')

print(f"Total de missões de sucesso: {missoes_sucesso}")
print(f"Total de missões de falha: {missoes_falha}")

Total de missões de sucesso: 3879
Total de missões de falha: 445


In [47]:
# Gráfico de pizza com o status das missões usando a coluna 'Mission Status'
fig_missao_status = px.pie(df, 
                            names='Mission Status', 
                            title='Status das Missões Espaciais',
                            color='Mission Status',
                            color_discrete_map={'Success': 'green', 'Failure': 'red'})
fig_missao_status.update_traces(textposition='inside', textinfo='percent+label')
fig_missao_status.show()

O gráfico de pizza acima oferece uma visão consolidada da proporção de sucessos e falhas em todas as missões. Ele revela que a vasta maioria das missões espaciais, 89.7%, foi bem-sucedida, enquanto apenas 10.3% resultaram em falha. Essa alta taxa de sucesso global ressalta a maturidade e a confiabilidade das tecnologias de lançamento, mesmo considerando a complexidade inerente de enviar objetos ao espaço. É um testemunho da dedicação e precisão exigidas na engenharia espacial ao longo das décadas desde o início da era espacial.

In [48]:
# Ano com a maior número de sucesso em relação ao total de lançamentos
ano_sucesso = df.groupby(df['Date'].dt.year)['Mission Status'].value_counts(normalize=True).unstack().fillna(0)
ano_sucesso['Success Ratio'] = ano_sucesso['Success'] / (ano_sucesso['Success'] + ano_sucesso['Failure'])
ano_sucesso = ano_sucesso.sort_values(by='Success Ratio', ascending=False)
print(f"O ano com o maior volume de Lançamentos Bem-Sucedidas em relação ao Total de Lançamentos foi {ano_sucesso.index[0]} com uma taxa de sucesso de {ano_sucesso['Success Ratio'].max():.2%}.")

# Ano com a menor taxa de sucesso
ano_falha = ano_sucesso.sort_values(by='Success Ratio', ascending=True)
print(f"O ano com o menor volume de Lançamentos Bem-Sucedidas em relação ao Total de Lançamentos foi {ano_falha.index[0]} com uma taxa de sucesso de {ano_falha['Success Ratio'].min():.2%}.")

# Transforma em porcentagem
ano_sucesso['Success Ratio'] = ano_sucesso['Success Ratio'] * 100

# Gráfico de linha com a taxa de sucesso ao longo dos anos
fig_sucesso_ano = px.line(
    ano_sucesso.sort_index(),
    x=ano_sucesso.sort_index().index,
    y='Success Ratio',
    title='Taxa de Sucesso das Missões Espaciais ao Longo dos Anos',
    labels={'x': 'Ano', 'Success Ratio': 'Taxa de Sucesso'}
)
fig_sucesso_ano.update_layout(xaxis_title='Ano', yaxis_title='Taxa de Sucesso')
fig_sucesso_ano.update_traces(line=dict(color='blue', width=2))
fig_sucesso_ano.show()


O ano com o maior volume de Lançamentos Bem-Sucedidas em relação ao Total de Lançamentos foi 1983 com uma taxa de sucesso de 98.48%.
O ano com o menor volume de Lançamentos Bem-Sucedidas em relação ao Total de Lançamentos foi 1958 com uma taxa de sucesso de 21.43%.


Este gráfico ilustra a evolução da taxa de sucesso das missões espaciais desde 1957. Observa-se que, nas primeiras décadas da exploração espacial (final dos anos 1950 e início dos anos 1960), a taxa de sucesso era consideravelmente baixa, com picos de apenas 20% em 1958, refletindo os desafios e a natureza experimental dos primeiros lançamentos. Contudo, houve um rápido aprendizado e aprimoramento tecnológico, levando a um aumento constante na taxa de sucesso, que se estabilizou em patamares elevados, geralmente acima de 90%, a partir dos anos 1970. Embora haja pequenas flutuações e algumas quedas pontuais (como em meados dos anos 80 e 90, que podem coincidir com acidentes notórios como o desastre do Challenger ou falhas de foguetes), a tendência geral é de alta confiabilidade nas últimas décadas, demonstrando a maturidade e o avanço da engenharia aeroespacial.

In [52]:
# Cria o DataFrame agrupado por ano e status da missão
lancamentos_ano_status = (
    df.groupby([df['Date'].dt.year, 'Mission Status'])
      .size()
      .reset_index(name='Quantidade')
      .rename(columns={'Date': 'Ano'})
)
lancamentos_ano_status.rename(columns={'Date': 'Ano'}, inplace=True)
lancamentos_ano_status['Ano'] = lancamentos_ano_status['Ano'].astype(int)

# Gráfico de barras agrupadas por ano e status da missão com cores customizadas
color_map = {'Failure': 'red', 'Success': 'green'}

fig_status_missao = px.bar(
    lancamentos_ano_status,
    x='Ano',
    y='Quantidade',
    color='Mission Status',
    color_discrete_map=color_map,
    barmode='group',
    title='Relação entre Lançamentos e Status das Missões por Ano',
    labels={'Ano': 'Ano', 'Quantidade': 'Número de Lançamentos', 'Mission Status': 'Status da Missão'}
)
fig_status_missao.update_layout(
    xaxis_title='Ano',
    yaxis_title='Número de Lançamentos',
    xaxis_tickangle=90,
    xaxis=dict(tickmode='linear')
)
fig_status_missao.show()


O gráfico apresenta a relação entre lançamentos espaciais e o status das missões (sucesso ou falha) por ano, de 1957 a 2020. Observa-se um aumento significativo no número de lançamentos a partir da década de 1960, com picos notáveis durante a corrida espacial (décadas de 1960 e 1970), especialmente entre 1975 e 1977, quando os lançamentos ultrapassaram 100 por ano, com predominância de missões bem-sucedidas (barras verdes). A partir da década de 1980, o número de lançamentos diminuiu e se manteve relativamente estável até o início dos anos 2010, quando houve novo crescimento, atingindo outro pico em 2018. Ao longo do tempo, percebe-se uma tendência clara de aumento da taxa de sucesso, com a quantidade de falhas (barras vermelhas) diminuindo significativamente, evidenciando o avanço tecnológico e a maior confiabilidade dos lançamentos espaciais ao longo das décadas.

In [ ]:
# Localidades com mais lançamentos bem-sucedidos
lancamentos_sucesso = df[df['Mission Status'] == 'Success']['Country'].value_counts().head(10)
# Gráfico de barras com os 10 países com mais lançamentos bem-sucedidos
fig_sucesso_pais = px.bar(
    lancamentos_sucesso,
    x=lancamentos_sucesso.index,
    y=lancamentos_sucesso.values,
    title='Top 10 Locais com mais Lançamentos Bem-Sucedidos',
    labels={'x': 'País', 'y': 'Quantidade de Lançamentos Bem-Sucedidos'}
)
fig_sucesso_pais.update_layout(
    xaxis_title='País',
    yaxis_title='Quantidade de Lançamentos Bem-Sucedidos'
)
fig_sucesso_pais.update_traces(texttemplate='%{y}', textposition='outside')
fig_sucesso_pais.show()

Observa-se que a Rússia e os EUA lideram com 1303 e 1186 lançamentos bem-sucedidos, respectivamente, o que reflete a vasta infraestrutura de lançamento que esses países construíram. O Cazaquistão, novamente, aparece em terceiro lugar com 608 lançamentos bem-sucedidos, reafirmando a importância do Cosmódromo de Baikonur como um dos locais mais ativos e bem-sucedidos na história dos lançamentos espaciais. A França e a China seguem com números significativos (285 e 243, respectivamente), mostrando a relevância de seus próprios locais de lançamento. A inclusão do "Oceano Pacífico", "New Zealand" e "Kenya" pode indicar locais de lançamento flutuantes ou instalações costeiras usadas para trajetórias específicas, ou ainda a emergência de novos players com menor volume, como a Nova Zelândia com o Rocket Lab.

In [ ]:
# Relaciona o status da missão com o Country
lancamentos_pais_status = df.groupby(['Country', 'Mission Status']).size().reset_index(name='Quantidade')

# Ordena os dados por quantidade de lançamentos
lancamentos_pais_status = lancamentos_pais_status.sort_values(by='Quantidade', ascending=False)

# Mostra apenas o Top 10
lancamentos_pais_status = lancamentos_pais_status.head(10)

# Cria um gráfico de barras empilhadas para visualizar o status das missões por país
fig_pais_status = px.bar(
    lancamentos_pais_status,
    x='Country',
    y='Quantidade',
    color='Mission Status',
    title='Status das Missões por País',
    labels={'Country': 'País', 'Quantidade': 'Número de Lançamentos', 'Mission Status': 'Status da Missão'},
    color_discrete_map={'Success': 'green', 'Failure': 'red'}
)
fig_pais_status.update_layout(
    xaxis_title='País',
    yaxis_title='Número de Lançamentos',
    xaxis_tickangle=45,
    barmode='stack'
)
fig_pais_status.update_traces(texttemplate='%{y}', textposition='outside')
fig_pais_status.show()

O gráfico "Status das Missões por País" complementa a análise anterior, mostrando a distribuição de sucessos e falhas por nação. A Rússia (ex-URSS) e os EUA, os dois maiores players históricos, apresentam o maior número de lançamentos bem-sucedidos (1303 e 1186, respectivamente) e também as maiores contagens de falhas (92 e 158, respectivamente). A proporção de falhas para os EUA é um pouco maior que a da Rússia em relação ao volume total de lançamentos, mas ambos os países demonstram uma alta taxa de sucesso. O Cazaquistão, embora não seja um país com capacidade primária de desenvolvimento de foguetes, possui um volume considerável de lançamentos a partir de seu território (608 sucessos e 93 falhas), novamente reforçando seu papel como uma importante base de lançamento. Países como França, China, Japão e Índia também exibem taxas de sucesso elevadas, consolidando suas posições como potências espaciais com capacidades de lançamento robustas

In [ ]:
# Relação entre o status da missão e a Empresa
lancamentos_empresa_status = df.groupby(['Company Name', 'Mission Status']).size().reset_index(name='Quantidade')
# Ordena os dados por quantidade de lançamentos
lancamentos_empresa_status = lancamentos_empresa_status.sort_values(by='Quantidade', ascending=False)
# Mostra apenas o Top 10
lancamentos_empresa_status = lancamentos_empresa_status.head(10)
# Cria um gráfico de barras empilhadas para visualizar o status das missões por empresa
fig_empresa_status = px.bar(
    lancamentos_empresa_status,
    x='Company Name',
    y='Quantidade',
    color='Mission Status',
    title='Status das Missões por Empresa',
    labels={'Company Name': 'Empresa', 'Quantidade': 'Número de Lançamentos', 'Mission Status': 'Status da Missão'},
    color_discrete_map={'Success': 'green', 'Failure': 'red'}
)
fig_empresa_status.update_layout(
    xaxis_title='Empresa',
    yaxis_title='Número de Lançamentos',
    xaxis_tickangle=45,
    barmode='stack'
)
fig_empresa_status.update_traces(texttemplate='%{y}', textposition='outside')
fig_empresa_status.show()

O gráfico detalha o sucesso e falha dos lançamentos por diferentes entidades. A RVSN URSS novamente se destaca com o maior número de lançamentos (total de 1777), dos quais 1614 foram sucessos e 163 foram falhas, indicando uma taxa de sucesso impressionante apesar do volume, mas também mostrando que mesmo a potência espacial tinha desafios. Empresas como Arianespace (269 sucessos), CASC (231 sucessos) e General Dynamics (203 sucessos) também demonstram altas taxas de sucesso, evidenciando a maturidade e a confiabilidade crescentes da indústria de lançamento espacial. A presença de falhas, mesmo em menor número, para a maioria das empresas, ressalta a complexidade inerente e os riscos associados às operações de lançamento espacial.

#### Em Relação ao Foguete

In [ ]:
# Em StatusRocket, remover a palavra Status
df['Status Rocket'] = df['Status Rocket'].str.replace('Status', '', regex=False)

# Gráfico de pizza com StatusRocket
fig_status_rocket = px.pie(
    df,
    names='Status Rocket',
    title='Status dos Foguetes',
    color='Status Rocket',
    color_discrete_map={
        'Active': 'blue',
        'Retired': 'gray',
        'Destroyed': 'red'
    }
)
fig_status_rocket.update_traces(textposition='inside', textinfo='percent+label')
fig_status_rocket.show()

O gráfico de pizza "Status dos Foguetes" oferece uma visão geral concisa do estado da frota de foguetes, indicando que uma vasta maioria, 81.7%, está "Aposentada" (Retired), enquanto apenas 18.3% estão "Ativos" (Active). Essa distribuição é esperada, dada a natureza de uma atividade que se estende por décadas desde 1957; a maioria dos foguetes construídos e lançados ao longo da história naturalmente não está mais em operação. Isso reforça a importância da manutenção e desenvolvimento contínuo de novas tecnologias de lançamento para sustentar a atividade espacial, e também reflete a desativação de tecnologias antigas e a transição para sistemas mais eficientes e modernos

#### Em relação ao Custo da Missão

In [ ]:
# Troca o nome da coluna 'Rocket'para 'Mission Cost'
df.rename(columns={' Rocket': 'Mission Cost'}, inplace=True)

# Converte a coluna 'Mission Cost' para numérica, removendo caracteres não numéricos	
df['Mission Cost'] = pd.to_numeric(df['Mission Cost'].str.replace(r'[^\d.]', '', regex=True), errors='coerce')

In [ ]:

# Gráfico de barras com o custo das missões espaciais por empresa
fig_mission_cost = px.bar(
    df.groupby('Company Name')['Mission Cost'].mean().sort_values(ascending=False).head(10),
    x=df.groupby('Company Name')['Mission Cost'].mean().sort_values(ascending=False).head(10).index,
    y=df.groupby('Company Name')['Mission Cost'].mean().sort_values(ascending=False).head(10).values,
    title='Custo Médio das Missões Espaciais por Empresa',
    labels={'x': 'Empresa', 'y': 'Custo Médio da Missão (USD)'}
)
fig_mission_cost.update_layout(
    xaxis_title='Empresa',
    yaxis_title='Custo Médio da Missão (USD)',
    yaxis_tickprefix='$',
    yaxis_tickformat=','
)
fig_mission_cost.update_traces(texttemplate='%{y:.2f}', textposition='outside')
fig_mission_cost.show()

Este gráfico de "Custo Médio das Missões Espaciais por Empresa" mostra uma disparidade gritante, com a RVSN URSS liderando com um custo médio de $5000 USD por missão, valor consideravelmente superior ao de qualquer outra entidade. Essa cifra elevadíssima para a RVSN URSS pode ser um reflexo dos vastos recursos investidos no desenvolvimento de foguetes e mísseis intercontinentais durante a Guerra Fria, onde a performance e a capacidade eram priorizadas sobre a economia, ou talvez uma inclusão de custos de desenvolvimento que não são comparáveis com outras empresas. A NASA aparece em segundo lugar, mas com um custo médio muito inferior (aproximadamente $511 USD), seguida por empresas como Boeing ($177 USD), Arianespace ($170 USD) e ULA ($151 USD). A JAXA (Agência de Exploração Aeroespacial do Japão) e a US Air Force também apresentam custos médios mais baixos, indicando que, fora o caso histórico da RVSN URSS, o custo médio por missão para as demais grandes agências e empresas é substancialmente menor e mais nivelado.

In [ ]:
# Gráfico de barras com o custo médio por pais
fig_cost_country = px.bar(
    df.groupby('Country')['Mission Cost'].mean().sort_values(ascending=False).head(10),
    x=df.groupby('Country')['Mission Cost'].mean().sort_values(ascending=False).head(10).index,
    y=df.groupby('Country')['Mission Cost'].mean().sort_values(ascending=False).head(10).values,
    title='Custo Médio das Missões Espaciais por País',
    labels={'x': 'País', 'y': 'Custo Médio da Missão (USD)'}
)
fig_cost_country.update_layout(
    xaxis_title='País',
    yaxis_title='Custo Médio da Missão (USD)',
    yaxis_tickprefix='$',
    yaxis_tickformat=','
)
fig_cost_country.update_traces(texttemplate='%{y:.2f}', textposition='outside')
fig_cost_country.show()


O gráfico indica que o Cazaquistão apresenta o custo médio mais alto por missão (aproximadamente $264 USD), seguido por EUA (aproximadamente $216 USD) e França (aproximadamente $171 USD). O elevado custo médio para o Cazaquistão, que abriga o Cosmódromo de Baikonur, sugere que as operações a partir deste local, frequentemente de grande porte e complexidade, contribuem para esse valor. A presença de "Gran Canaria" e "Pacific Missile Range Facility" pode se referir a instalações de rastreamento ou apoio a missões, não necessariamente países lançadores. Notavelmente, Rússia e China, apesar de serem grandes players em número de lançamentos, aparecem com custos médios significativamente mais baixos ($40.54 e $40.27, respectivamente), o que pode refletir metodologias de custo diferentes, uma maior padronização de lançadores ou uma política de preços distinta.

#### Em relação a Empresa

In [ ]:
# Gráfico de barrar com o número de lançamentos por empresa
fig_lancamentos_empresa = px.bar(
    df['Company Name'].value_counts().head(10),
    x=df['Company Name'].value_counts().head(10).index,
    y=df['Company Name'].value_counts().head(10).values,
    title='Top 10 Empresas com mais Lançamentos Espaciais',
    labels={'x': 'Empresa', 'y': 'Número de Lançamentos'}
)
fig_lancamentos_empresa.update_layout(
    xaxis_title='Empresa',
    yaxis_title='Número de Lançamentos',
)
fig_lancamentos_empresa.update_traces(texttemplate='%{y}', textposition='outside')
fig_lancamentos_empresa.show()


O gráfico "Top 10 Empresas com mais Lançamentos Espaciais" destaca a esmagadora contribuição histórica da RVSN URSS (Forças de Foguetes Estratégicos da União Soviética) com 1777 lançamentos, um número muito superior ao de qualquer outra entidade, reiterando o papel central da União Soviética na corrida espacial e na exploração inicial. A grande lacuna para a segunda colocada, Arianespace (279), e outras empresas como General Dynamics (251) e CASC (251), demonstra a magnitude do esforço espacial soviético. A presença de entidades militares como a US Air Force (161) e a VKS RF (201) (Forças Aeroespaciais Russas) sublinha a natureza frequentemente dual, civil e militar, das operações espaciais. A NASA (203) e empresas aeroespaciais tradicionais como Boeing (136), ULA (140) e Martin Marietta (114) também aparecem, mostrando a diversidade de atores que contribuíram para a história dos lançamentos espaciais.

In [ ]:
# Gráfico de barras com o número de foguetes ativos por país
fig_ativos_pais = px.bar(
    df[df['Status Rocket'] == 'Active']['Country'].value_counts().head(10),
    x=df[df['Status Rocket'] == 'Active']['Country'].value_counts().head(10).index,
    y=df[df['Status Rocket'] == 'Active']['Country'].value_counts().head(10).values,
    title='Top 10 Países com mais Foguetes Ativos',
    labels={'x': 'País', 'y': 'Número de Foguetes Ativos'}
)
fig_ativos_pais.update_layout(
    xaxis_title='País',
    yaxis_title='Número de Foguetes Ativos',
)
fig_ativos_pais.update_traces(texttemplate='%{y}', textposition='outside')
fig_ativos_pais.show()

Este gráfico revela uma mudança notável na liderança espacial atual em comparação com o histórico de foguetes aposentados. Atualmente, a China lidera com 223 foguetes ativos, superando os EUA (208) e a França (113). Essa dominância chinesa reflete seus investimentos massivos e o rápido avanço de seu programa espacial nas últimas décadas, posicionando-a como uma força crescente no cenário espacial global. A presença do "Oceano Pacífico" e "Nova Zelândia" na lista sugere o papel de locais de lançamento ou de pequenas nações que estão ganhando relevância em nichos específicos, como o lançamento de pequenos satélites, muitas vezes por meio de empresas privadas. A Rússia, que dominava a contagem de foguetes aposentados, aparece mais abaixo na lista de ativos (36), indicando uma reconfiguração do cenário de poder espacial, onde a China e os EUA se destacam na vanguarda das operações ativas.

In [ ]:
# Gráfico de barras com o número de foguetes aposentados por país
fig_aposentados_pais = px.bar(
    df[df['Status Rocket'] == 'Retired']['Country'].value_counts().head(10),
    x=df[df['Status Rocket'] == 'Retired']['Country'].value_counts().head(10).index,
    y=df[df['Status Rocket'] == 'Retired']['Country'].value_counts().head(10).values,
    title='Top 10 Países com mais Foguetes Aposentados',
    labels={'x': 'País', 'y': 'Número de Foguetes Aposentados'}
)
fig_aposentados_pais.update_layout(
    xaxis_title='País',
    yaxis_title='Número de Foguetes Aposentados',
)   
fig_aposentados_pais.update_traces(texttemplate='%{y}', textposition='outside')
fig_aposentados_pais.show()

No gráfico acima, observa-se 